src/daml/Daml/Trading/Reward/TradingPoint.daml

```diff
@@ -1 +1,164 @@
 module Daml.Trading.Reward.TradingPoint where 
+
+import Daml.Script
+                
+template Trade_Success
+    with
+        admin : Party
+        seller : Party
+        buyer : Party
+        tradeAmount : Decimal
+        asset : Text
+       
+    where 
+        signatory admin
+        observer seller, buyer
+        ensure seller /= buyer
+        choice IssueRewardPoint : (ContractId TradingPoint, ContractId TradingPoint)
+            controller admin
+            do  
+                (sellerTradingPointCid, sellerTradingPoint) <- fetchByKey @TradingPoint (admin, seller)
+                (buyerTradingPointCid, buyerTradingPoint) <- fetchByKey @TradingPoint (admin, buyer)
+                let 
+                    oldSellerTradingPoint = sellerTradingPoint.quantity
+                    oldBuyerTradingPoint = buyerTradingPoint.quantity
+
+                    oldSellerTradeAmountRemain = sellerTradingPoint.tradeAmountRemain
+                    oldBuyerTradeAmountRemain = buyerTradingPoint.tradeAmountRemain
+
+                    quantityPointBaseTradeAmount = truncate (tradeAmount / 500.0)
+                    tradeAmountRemainBaseTradeAmount = tradeAmount - intToDecimal(500 * truncate (tradeAmount / 500.0))
+                    
+                    newSellerTradeAmountRemain = tradeAmountRemainBaseTradeAmount + oldSellerTradeAmountRemain
+                    newBuyerTradeAmountRemain = tradeAmountRemainBaseTradeAmount + oldBuyerTradeAmountRemain
+                    
+                    sellerQuantityPointAfterCalAmountRemain = oldSellerTradingPoint + quantityPointBaseTradeAmount + truncate (newSellerTradeAmountRemain / 500.0) 
+                    buyerQuantityPointAfterCalAmountRemain = oldBuyerTradingPoint + quantityPointBaseTradeAmount + truncate (newBuyerTradeAmountRemain / 500.0)
+
+                    sellerQuantity = sellerQuantityPointAfterCalAmountRemain
+                    buyerQuantity = buyerQuantityPointAfterCalAmountRemain
+
+                    sellerAmountRemain = newSellerTradeAmountRemain - intToDecimal(500 * truncate (newSellerTradeAmountRemain / 500.0))
```

Comment:
we should create a function that receives `tradeAmount` as input and returns `(point, tradeAmountRemain)` instead of keep reusing the formula like this